pkg/controller/pod.go

```diff
@@ -2396,3 +2430,30 @@ func setPodRoutesAnnotation(annotations map[string]string, provider string, rout
 
 	return nil
 }
+
+func (c *Controller) getNadInterfaceFromNetworkStatusAnnotation(networkStatus string, nadName string) (string, error) {
```

Comment:
This can be moved to `util/network_attachment.go` and use the [NetworkStatus Type](https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/blob/master/pkg/apis/k8s.cni.cncf.io/v1/types.go#L105C6-L105C19)